### PR TITLE
[Merged by Bors] - fix(Tactic/Cases): fix context for induction'

### PR DESCRIPTION
**Pull request by joneugster**
Adding a test case for `induction'` that failed and fixing it: `induction' a` would accidentally leave `a` and variables depending on `a` in context.

---

This is the example that's been added to the test cases:

```lean
example (a b : ℕ) (h : a + b = a) : b = 0 := by
  induction' a with d hd
  · sorry
  · -- bad context:
    -- a b: ℕ
    -- h✝: a + b = a
    -- d: ℕ
    -- hd: d + b = d → b = 0
    -- h: Nat.succ d + b = Nat.succ d
    -- ⊢ b = 0
    sorry
```

See [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60induction'.60.20leaving.20junk)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
bors d+

File: test/propose.lean [hunk 1]
```diff
@@ -73,8 +73,6 @@ info: Try this: have : p ≠ 1 := ne_one hp
 ---
 info: Try this: have : p ∣ p * p ↔ p ∣ p ∨ p ∣ p := dvd_mul hp
 ---
-info: Try this: have : p ∣ a := Prime.dvd_of_dvd_pow hp h
----
```
Comment (joneugster) [inline test/propose.lean:77]: I don't quite understand why this change to `induction'` weakens this test for `have?!` slightly. At least the dropped suggestion is not one of the two important ones.

@semorrison should this be looked at before merging?

Comment (joneugster) [reply test/propose.lean:77]: Ah nvm, this is fine. After starting induction, the hypothesis is now `h : p ∣ 1` and one would need to turn it into `h : p ∣ a ^ 0` before being able to use `Prime.dvd_of_dvd_pow`. And that's not something `have?!` is supposed to figure out by itself, for one because `a` is ambiguous in that step.

**Comment by semorrison** [comment]
@joneugster do you want to merge master, and see if there is a problem, and then send it to bors again?
